docs/backend_api.rst

```diff
@@ -5,7 +5,7 @@ Reference simulator
 .. default-role:: obj
 
 Nengo is designed so that models created with the
-:doc:`Nengo modeling API <frontend_api>`
+:doc:`Nengo core API <core_api>`
```

Comment:
I don't love the name "Nengo core", mainly because I don't think "core" really indicates what users should expect to find there.  E.g., is the backend less "core" than the frontend?  I also think we have been using "Nengo core" in other places to refer more generally to the frontend+default nengo backend.

Personally I thought "Nengo modeling API" was pretty clear, but of course that's all subjective :smile:.  If we want something different, perhaps we could just use the frontend/backend language directly (i.e. "Nengo frontend API")?

Reply:
I picked 'nengo core' because it's what's in this picture: https://www.nengo.ai/projects.html and in that context is referring only to the frontend API, I think.

I'm definitely not wedded to it, though.  The frontend/backend would be better if we plan on having separate docs for both, for instance.  Mainly I wanted to avoid calling it 'frontend api' in once place (the url) and 'modelling api' in a different place (the menu link).

Reply:
> I picked 'nengo core' because it's what's in this picture: https://www.nengo.ai/projects.html and in that context is referring only to the frontend API, I think.

Yeah that's true.  Although right below that, we have "Core frameworks" which includes a bunch of stuff not in what we'd call "Nengo core", and then inside "Core frameworks" we have "Nengo core" which includes the reference simulator.  So we might be overloading the word "core" there :wink:

> The frontend/backend would be better if we plan on having separate docs for both, for instance. 

We do have a backend_api page (describing the simulator api), although it's not labeled as such (https://www.nengo.ai/nengo/backend_api.html). So we could apply the frontend/backend terminology there if we want.

> Mainly I wanted to avoid calling it 'frontend api' in once place (the url) and 'modelling api' in a different place (the menu link).

I don't think it's a general expectation that menu headings match up with URLs.  E.g. if we look at https://www.tensorflow.org/programmers_guide/ (just the first thing I looked at since I had the tab open), a lot of the left-hand links direct to pages with URLs that don't match the menu heading.  Which isn't to say that we shouldn't have a more informative name, if we think "Nengo modeling API" isn't clear.  But I don't think we need to constrain ourselves to "what would be an informative name AND a good URL".  Or we could change the page to "modeling_api.html" if we like that name but want to have the names match.